### PR TITLE
Add LegislativePeriod#unique_people method

### PR DESCRIPTION
**Pull request by chrismytton**
This returns a list of the unique people for a legislative period by
discarding any CSV rows which have the same id.

### CODE REVIEW

File: lib/everypolitician.rb [hunk 1]
```diff
@@ -145,12 +145,16 @@ def csv_url
...
-      CSV.parse(open(csv_url).read, headers: true, header_converters: :symbol, converters: nil)
+      @csv ||= CSV.parse(open(csv_url).read, headers: true, header_converters: :symbol, converters: nil)
     end
 
     def [](key)
       raw_data[key]
     end
+
+    def unique_people
+      csv.map(&:to_hash).uniq { |person| person[:id] }
```
Comment (tmtmtmtm) [inline lib/everypolitician.rb:156]: I'm a little worried that there's still a mismatch between what's being asked for here, and what's being returned, that could be confusing in some cases, and possibly bug-inducing in others. 

e.g. if someone starts asking any of these 'People' for things like a `start_date` or `end_date` odd things will happen.

**Comment by tmtmtmtm** [comment]
I'm wondering if this is the wrong way to think about the issue. 

My understanding is that this is required because the rows of the CSV are Memberships, but often you want to work with People instead. 

So rather than a `.unique_people` which subtly implies that you're already dealing with People, but possibly with duplicates, we want a method that more clearly is about converting the type of things in the list?

**Comment by chrismytton** [comment]
@tmtmtmtm You're right, the `LegislativePeriod#unique_people` method has a bit of a misleading/confusing name.

Perhaps a `LegislativePeriod#people` method would be better? Then it could either:
1. Return the list of the unique memberships OR
2. Return an array of `Everypolitician::Popolo::Person` objects by using the unique ids from the csv behind the scenes. This might be preferable as then they can call `.start_date`, `.end_date` etc on the returned object.

**Comment by tmtmtmtm** [comment]
If you're working with [Eduskunta 36](http://everypolitician.org/finland/eduskunta/term-table/36.html), and get a list of people from that, what would you expect the `.start_date` and `.end_date` methods to return for Jyrki Yrttiaho?

**Comment by tmtmtmtm** [comment]
@chrismytton is this something you're going to pick up again, or should we just discard it?

BTW I'm a little confused why this is something you'd want explicitly implemented here, rather than on https://github.com/everypolitician/everypolitician-popolo/ …

**Comment by chrismytton** [comment]
This was the wrong approach, as you say if we do end up needing it everypolitician-popolo is the place for it.

**Pull request closed without merging**